binnurr/assignment-3/TrieNode.java

```
@@ -0,0 +1,41 @@
+/**
+ * A Trie node has children nodes as many as number of characters in the alphabet. Trie node has a
+ * field isWord to indicate if the route from root to that node corresponds to a valid word.
+ */
+public class TrieNode {
+    private final int alphabetSize = 26;
+    private TrieNode[] children;
+    private boolean isWord;
+
+    public TrieNode(){
+        this.children = new TrieNode[alphabetSize];
+        this.isWord = false;
+    }
+
+    /** Returns if the node is word or not. */
+    public boolean getIsWord() {
+        return this.isWord;
+    }
+
+    /** Return the corresponding child node of the node for the given character. */
+    public TrieNode getChild(char character) {
+        char upperCaseCharacter = Character.toUpperCase(character);
+        if (upperCaseCharacter >= 'A' && upperCaseCharacter <= 'Z') {
+            return this.children[character - 'A'];
+        }
+        return null;
+    }
+
+    /** Sets if the node is word. */
+    public void setIsWord(boolean isWord) {
+        this.isWord = isWord;
+    }
+
+    /** Constructs the corresponding child node of the node for the given character. */
+    public void setChild(char character) {
+        char upperCaseCharacter = Character.toUpperCase(character);
+        if (upperCaseCharacter >= 'A' && upperCaseCharacter <= 'Z') {
```

Currently if this is not true, you just ignore it, but it's better to throw an exception instead.
One way to do this is to use checkArgument, similarly to checkNotNull
```
import static com.google.common.base.Preconditions.checkArgument;

checkArgument(upperCaseCharacter >= 'A' && upperCaseCharacter <= 'Z', 'Character should be upper case letter');
```

---

perfect, thanks!